messageID.go

```diff
@@ -11,20 +11,17 @@ type IDGenerator interface {
 // concurrent use by multiple goroutines.
 func NewSafeID(startID int) IDGenerator {
 	return &safeID{
-		nextID: startID,
-		mutex:  &sync.Mutex{},
+		nextID: int64(startID),
 	}
 }
 
 type safeID struct {
-	nextID int
-	mutex  *sync.Mutex
+	nextID int64
 }
 
-func (s *safeID) Next() int {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-	id := s.nextID
-	s.nextID++
+func (s *safeID) Next() (id int) {
+	id = int(atomic.LoadInt64(&s.nextID))
+	atomic.AddInt64(&s.nextID, 1)
```

Comment:
this should at least be safe and faster, as it's only one operation:  
```go
func (s *safeID) Next() int {
        id := atomic.AddInt64(&s.nextID, 1)

        return int(id)
}
```

Reply:
good catch. will fix.